include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -218,7 +218,11 @@ struct transform_reduce
         const _Size __adjusted_global_id = __global_offset + __iters_per_work_item * __global_idx;
         const _Size __adjusted_n = __global_offset + __n;
         // Sequential load and reduce from global memory
-        union __storage {_Tp __v; __storage(){} } __res;
+        union __storage
+        {
+            _Tp __v;
+            __storage() {}
```

Comment:
`__storage() = default;`   ?

Reply:
And is this constructor really needed?

Reply:
This approach is consistent with the existing code base. https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h#L169

If you wish to change this, I would recommend refactoring it in a separate PR